### DISCUSSION TITLE
FYI - If you're having issues with Surveillance Station streams, check the user permissions in Surveillance Station

### DISCUSSION THREAD

**Discussion by jordancrawford** [koush/scrypted#1456] [General]
I just want to share something I've learnt the hard way in case it helps others.

I use Scrypted in Docker on my Synology NAS, pointed at my camera through Synology Surveillance Station.
I authenticate Scrypted through a non-admin user (to reduce the risk if my credentials ever get leaked somehow).

I was experiencing a weird issue where stream snapshots work fine, but videos streams (though HomeKit and the Scrypted web UI) don't work. In the logs I would see the following message:

```
[Driveway] 1 prebuffer session ended with error RPCResultError [SynologyApiError]: Insufficient user privilege (error code 105)
    at a.SynologyApiClient.sendRequest (/plugin/..\src\api\synology-api-client.ts:170:23)
    at processTicksAndRejections (node:internal/process/task_queues:95:5)
    at a.SynologyApiClient.liveViewPaths [as getCameraLiveViewPath] (/plugin/..\src\api\synology-api-client.ts:54:9)
    at l.getVideoStream (/plugin/..\src\main.ts:86:19)
    at RpcPeer.handleMessageInternal (/server/node_modules/@scrypted/server/src/rpc.ts:759:37)
@scrypted/synology-ss:host
host:@scrypted/prebuffer-mixin
host:@scrypted/prebuffer-mixin {
```

I confirmed I could use the same credentials to login to the Synology web UI and view the camera through there.
It also worked fine with my main user (the one which is administrator of the camera).
It wasn't making sense why some API calls worked (to get the snapshot) but others didn't (to get the streams).

It turns out that in addition to the DSM user settings for which applications a user can access, you also need to ensure the permissions are set correctly in the "User" section within Surveillance Station. In the "Edit" -> "Privilege" menu I set my Scrypted user to the "All-Object-Manager".

<img width="989" alt="Screenshot 2024-05-01 at 10 18 13 PM" src="https://github.com/koush/scrypted/assets/7520773/efadd15a-c2cd-44c2-97dd-4c8e366825df">

Also - I noticed that my session (or perhaps the stream URL) was sometimes cached, meaning if I changed the credentials to a broken user for testing, the stream would continue to work. This gave me a false sense of the problem. While debugging, I recommend hitting "reload plugin" after changing the credentials to ensure it's really using the credentials provided.

**Comment by rAf19** [comment]
@jordancrawfordnz Thank you very much!  Literally been having this issue since May and I just came accross your comment and it worked BUT I did have to roll back the Rebroadcast Plugin to 0.10.16 as the stream does not work in the Homekit app.  

Here is what the console says for one my cams when in the latest Rebroadcast plugin 0.10.18:

Rebroadcast Plugin]: 
[Rebroadcast Plugin]: rebroadcast error O [Error]: RTSP Error: RTSP/1.0 404 Stream Not Found
[Rebroadcast Plugin]:     at t.RtspClient.request (/@scrypted/prebuffer-mixin/main.nodejs.js:1:87674)
[Rebroadcast Plugin]:     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
[Rebroadcast Plugin]:     at async Object.P [as start] (/@scrypted/prebuffer-mixin/main.nodejs.js:1:141789) {
[Rebroadcast Plugin]:   status: {
[Rebroadcast Plugin]:     line: 'RTSP/1.0 404 Stream Not Found',
[Rebroadcast Plugin]:     code: 404,
[Rebroadcast Plugin]:     version: 'RTSP/1.0',
[Rebroadcast Plugin]:     reason: 'Stream'
[Rebroadcast Plugin]:   }
[Rebroadcast Plugin]: }
[Rebroadcast Plugin]: 1 prebuffer session ended
[Rebroadcast Plugin]: Garage restarting prebuffer session in 5 seconds
[Rebroadcast Plugin]: rtsp outgoing message
[Rebroadcast Plugin]:  TEARDOWN rtsp://10.86.1.38:554/Sms=4.unicast RTSP/1.0
[Rebroadcast Plugin]: CSeq: 6
[Rebroadcast Plugin]: User-Agent: Scrypted
[Rebroadcast Plugin]: Authorization: Digest username="syno", realm="LIVE555 Streaming Media", nonce="d5869ded6f48a0ca45f0a96df00e1457", uri="rtsp://10.86.1.38:554/Sms=4.unicast", algorithm="MD5", response="70a0ce72bc313e6ea1b44fe2191f1327"
[Rebroadcast Plugin]: Session: CAD

**Reply by jordancrawford** [reply]
Hi @rAf19 - I'm glad that helped.

I had this issue too - see #1391
I followed koush's instructions from the comments and it fixed my issue.

**Comment by rAf19** [comment]
I was able to use ffmpeg parser (TCP) for each stream and now it works using the latest rebroadcast plugin!
Much appreciated my good sir!
